compiler/plugins/target/ROCM/Dialect/ROCM/IR/ROCMBuiltinManager.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "compiler/plugins/target/ROCM/Dialect/ROCM/IR/ROCMDialect.h"
+#include "mlir/Parser/Parser.h"
+
+namespace mlir::iree_compiler::IREE::ROCM {
+
+FailureOr<ModuleOp> ROCMDialect::getOrLoadBuiltinModule(StringRef path) {
+  std::optional<StringRef> maybeBuiltin = builtins.getFile(path);
+  if (!maybeBuiltin) {
+    return failure();
+  }
+
+  // Acquire a lock on the map that will release once out of scope.
+  std::lock_guard<std::mutex> guard(builtinMutex);
+  MLIRContext *ctx = getContext();
+
+  auto loadedLibrary = builtinModules.find(path);
+  if (loadedLibrary != builtinModules.end()) {
+    // Check whether the library already failed to load.
+    if (ModuleOp module = loadedLibrary->second.get()) {
+      return module;
+    }
+    return failure();
+  }
+
+  // We update the storage for the library regardless of whether parsing
+  // succeeds so that other threads don't have to retry.
+  OwningOpRef<ModuleOp> &parsedLibrary = builtinModules[path];
+
+  parsedLibrary = parseSourceString<mlir::ModuleOp>(maybeBuiltin.value(), ctx);
```

Comment:
FYI, this also works
```suggestion
  parsedLibrary = parseSourceString<mlir::ModuleOp>(*maybeBuiltin, ctx);
```

(but either is fine)

Reply:
I started preferring `.value()` because it's clearer to me that it's an optional rather than any old pointer. Not enough to ask on review though (and will go with whatever style is requested).